### PR TITLE
feat: add mykg_get_source MCP tool for bounded source-text excerpts

### PR DESCRIPTION
**Pull request by SenolIsci**
## Summary
- Adds a new `mykg_get_source` MCP tool that resolves source-text excerpts for a node, an edge, or both, using `chunk_node_index.json` + `file_manifest.json`
- Narrows each source chunk down to the passages actually mentioning the relevant name(s) instead of returning the full ~2000-token chunk
- For edges, prefers chunks where both endpoints co-occur (`exact_chunk_overlap: true`); falls back to the union of endpoint chunks otherwise
- Adds `edges_by_id`, `file_manifest`, and `chunk_node_index_by_id` indexes to `KnowledgeGraph`, loaded/rebuilt on `reload()`

## Test plan
- [x] `uv run pytest tests/test_mcp_server.py -q` — 40 passed
- [ ] Manual MCP smoke test: call `mykg_get_source` with a known node_id / edge_id against a real session and confirm excerpts are returned

## Summary by Sourcery

Add a new MCP tool to retrieve bounded source-text excerpts for nodes and edges, backed by new chunk and file manifest indexes in the knowledge graph.

New Features:
- Introduce the mykg_get_source MCP tool that returns focused source-text excerpts for a node, an edge, or both.

Enhancements:
- Extend KnowledgeGraph with indexes for edges by ID and chunk-to-node mappings derived from chunk_node_index.json and file_manifest.json.
- Load and reuse intermediate chunk and file manifest artifacts when initializing or reloading knowledge graph sessions.

Tests:
- Add fixture-backed tests covering node and edge excerpt resolution, overlap vs union behavior, truncation by max_chunks, and missing chunk artifact handling.
- Update MCP server tool registration test to include the new mykg_get_source tool and adjusted tool count.

### CODE REVIEW

File: tests/test_mcp_server.py [hunk 1]
```diff
@@ -340,10 +382,73 @@ def test_mcp_server_has_tools(self):
...
+        # organization-acme is only in partners.md::1 — no overlap.
+        result = _resolve_edge_excerpt(kg_with_chunks, "edge-001", max_chunks=3)
+        assert "error" not in result
+        assert result["exact_chunk_overlap"] is False
+        assert result["total_chunks"] == 2
+        sources = {e["source_file"] for e in result["excerpts"]}
+        assert sources == {"team.md", "partners.md"}
+
+    def test_edge_excerpt_unknown_edge(self, kg_with_chunks: KnowledgeGraph):
+        result = _resolve_edge_excerpt(kg_with_chunks, "nonexistent", max_chunks=3)
```
[the remark this answers is not in the document]
Comment (SenolIsci) [reply tests/test_mcp_server.py:484]: Addressed in 735e91d — the unknown-edge error now echoes the missing `edge_id` and points to `mykg_query_graph`/`mykg_search_nodes`, matching the unknown-node error's format. Test updated to assert `nonexistent` appears in the error message.

File: tests/test_mcp_server.py [hunk 2]
```diff
@@ -113,6 +119,42 @@ def kg(session_dir: Path) -> KnowledgeGraph:
     return KnowledgeGraph(session_root=session_dir)
 
 
+TEAM_MD_TEXT = (
+    "Alice Smith is a software engineer at Acme Corp. "
+    "Bob Jones manages the infrastructure team at Acme Corp."
```
[the remark this answers is not in the document]
Comment (SenolIsci) [reply tests/test_mcp_server.py:128]: Good suggestion, deferring for now — this would mean redesigning the fixtures (overlapping mentions, deliberately long text to exercise max_total truncation) rather than a small fix, and the current tests already cover the core contract (name presence, excerpt bounded by source length). Will pick this up in a follow-up if we see excerpt-boundary bugs in practice.

**Comment by SenolIsci** [comment]
Thanks for the review — addressed in 735e91d:

1. **`_resolve_edge_excerpt` set precompute** — done, `to_set = set(to_chunks)` is now built once before the comprehension.
2. **Silent `JSONDecodeError`/`OSError` swallowing in `_load_chunk_artifacts`** — done, now logs a `_log.warning(..., exc_info=True)` with the path, so malformed/missing intermediate files are diagnosable instead of silently disabling source-chunk lookup.
3. **Caching `build_chunk_texts(kg.file_manifest)`** — deferred. It's an in-memory dict comprehension over already-loaded JSON, not file I/O, so the per-call cost looked low enough not to justify adding a cache-invalidation surface on `KnowledgeGraph.reload()`. Happy to revisit if profiling on a large session shows otherwise.

Replied inline on the two test-coverage suggestions as well.
